backend/models/form_response.py

```diff
@@ -14,6 +15,18 @@ class FormResponse(BaseModel):
     antispam: t.Optional[AntiSpam]
     response: dict[str, t.Any]
     form_id: str
+    timestamp: str
+
+    @validator("timestamp", pre=True)
+    def set_timestamp(cls, iso_string: str) -> str:
```

Comment:
```suggestion
    def set_timestamp(cls, iso_string: t.Optional[str]) -> t.Optional[str]:
```
You do `None` check, so `iso_string` can be `None`. When this raise `ValueError`, then no string is returned and this is also `None`.

Reply:
Good catch, thanks.